maven/bnd-reporter-maven-plugin/src/main/java/aQute/bnd/maven/reporter/plugin/ReadmeMojo.java

```diff
@@ -0,0 +1,153 @@
+	private MavenProject		project;
+
+	@Parameter(defaultValue = "${session}", readonly = true, required = true)
+	private MavenSession		session;
+
+	@Parameter(property = "bnd.reporter.skip", defaultValue = "false")
+	private boolean				skip;
+
+	@Parameter
+	private Map<String, String>	parameters	= new HashMap<>();
+
+	@Override
+	public void execute() throws MojoExecutionException, MojoFailureException {
+		if (skip) {
+			logger.debug("skip project as configured");
+			return;
+		}
+
+		try (Processor processor = new Processor()) {
+			processor.setTrace(logger.isDebugEnabled());
+			processor.setBase(project.getBasedir());
+
+			MavenProjectWrapper toAnalyze = new MavenProjectWrapper(session.getProjects(), project);
+
+			// Set the readme configuration to the processor
+			processor.setProperty("-exportreport.__autoreadme",
+				"readme.md;template=default:readme.twig" + getParameters());
+
+			// Create the generator service.
+			ReportGeneratorService generator = null;
+			String scope = null;
+			if (toAnalyze.isAggregator()) {
+				generator = ReportGeneratorFactory.create()
+					.setProcessor(processor)
+					.useCustomConfig()
+					.withAggregatorProjectDefaultPlugins()
+					.build();
+			} else {
+				generator = ReportGeneratorFactory.create()
+					.setProcessor(processor)
+					.useCustomConfig()
+					.withProjectDefaultPlugins()
+					.build();
+			}
+
+			// Create the exporter service.
+			ReportExporterService reporter = ReportExporterBuilder.create()
+				.setProcessor(processor)
+				.setGenerator(generator)
+				.build();
+
+			logger.info("Generating reports...");
+
+			Map<String, Resource> reportResults = reporter.exportReportsOf(toAnalyze);
+
+			report(processor);
+
+			if (!processor.isOk()) {
+				throw new MojoExecutionException("Errors in bnd processing, see log for details.");
+			}
+
+			if (reportResults.isEmpty()) {
+				logger.info("No report matching the '{}' scope has been found.", scope);
+			}
+
+			for (Entry<String, Resource> result : reportResults.entrySet()) {
+				try {
+					result.getValue()
+						.write(new FileOutputStream(result.getKey()));
+					logger.info("The report at {} has been successfully created.", result.getKey());
+				} catch (Exception exception) {
+					throw new MojoExecutionException("Failed to write the report at " + result.getKey(), exception);
+				}
+			}
+		} catch (MojoExecutionException e) {
+			throw e;
+		} catch (Exception e) {
+			throw new MojoExecutionException("bnd error: " + e.getMessage(), e);
+		}
+	}
+
+	private void report(Processor processor) {
+		for (String warning : processor.getWarnings()) {
+			logger.warn("Warning : {}", warning);
+		}
+		for (String error : processor.getErrors()) {
+			logger.error("Error   : {}", error);
+		}
+	}
+
+	private String getParameters() {
+		Map<String, String> params = new HashMap<>(parameters);
+
+		System.getProperties()
+			.stringPropertyNames()
+			.stream()
+			.filter(k -> k.startsWith("reporter."))
```

Comment:
May use "bnd.reporter." like at the "bnd.reporter.skip" parameter

Reply:
Done